go.mod

```diff
@@ -4,6 +4,7 @@ go 1.24.0
 
 require (
 	github.com/stretchr/testify v1.11.1
+	github.com/testcontainers/testcontainers-go/modules/k3s v0.38.0
```

Comment:
Can we create a self contained module instead of adding test container dependency here ?

Reply:
@harshanarayana Thank you, do you want me to create a self contained module in `examples/testcontainers/` folder with its own go.mod and go.sum, so that the main `go.mod` will not be polluted ? Want to clarify this because I could not find a dedicated `go.mod` file in any of the existing examples and looking at `./hack/test-go.sh` such test will not be executed ?

Reply:
Yes. Self contained go module. 

Xref: https://github.com/kubernetes-sigs/e2e-framework/tree/main/examples%2Fcontroller_tests%2Ftestdata

Reply:
Thanks, a self contained module has been created.